src/clj/rems/application/commands.clj

```diff
@@ -408,36 +431,62 @@
 (defmethod command-handler :application.command/approve
   [cmd application injections]
   (or (entitlement-end-not-in-future-error (:entitlement-end cmd))
-      (add-comment-and-attachments cmd injections
+      (add-comment-and-attachments cmd application injections
                                    (merge {:event/type :application.event/approved}
                                           (when-let [end (:entitlement-end cmd)]
                                             {:entitlement/end end})))))
 
+(defn- empty-redacted-attachments-error [cmd]
+  (when (empty? (:redacted-attachments cmd))
+    {:errors [{:type :empty-redacted-attachments}]}))
+
+(defn- find-original-attachment-event-ids
+  "Finds events from which the redacted attachments originate from,
+   and returns those events ids."
+  [redacted-attachments application]
+  (let [events (:application/events application)
+        redacted-ids (set (map :attachment/id redacted-attachments))]
+    (vec (for [event events
+               :let [event-attachment-ids (set (map :attachment/id (:event/attachments event)))]
+               :when (seq (clojure.set/intersection redacted-ids event-attachment-ids))]
+           (select-keys event [:event/id])))))
```

Comment:
hmm, i see what you mean. it makes more sense from the model side of view to include redacted attachment id with the originating event id. however the context seeps unfortunately into the highlighter mechanism, which is dependent on event data. one solution could be to add `:application/request-ids` which accepts a list of related ids, and map the request ids into redacted event. however this does not take into account events that have no generated request ids, for example remarked event.

Reply:
I think the highlighter very likely should do different things depending on what the event is. 
- So for a review request or response, it could go through all the events and pick the ones that have the same request id. 
- For the redaction event, it should highlight the event where the attachment came from. 

I think we should have all this data available in the UI, so we can do it in the UI. There are also only "few" events ever (like 100 or so at max) so it should be workable? I don't think we need to pack this stuff somehow into the events or the read model?